sample-kotlin-koin/src/main/java/com/infullmobile/android/infullmvp/sample/fragment/SampleFragmentPresenter.kt

```diff
@@ -0,0 +1,18 @@
+package com.infullmobile.android.infullmvp.sample.fragment
+
+import android.os.Bundle
+import com.infullmobile.android.infullmvp.InFullMvpPresenter
+import com.infullmobile.android.infullmvp.sample.models.SharedPreferencesModel
+
+class SampleFragmentPresenter(view: SampleFragmentView,
+                              private val sharedPreferencesModel: SharedPreferencesModel)
+    : InFullMvpPresenter<SampleFragmentView>(view) {
+
+    override fun bind(extras: Bundle?, savedInstanceState: Bundle?) {
+        view.updateText(extras!!.getString(TEXT_EXTRA_KEY)!!)
```

Comment:
Yelling at variables is not very nice. If we have a way to pass null we should handle it.
Two options I see are `extras?.getString(TEXT_EXTRA_KEY) ?: ""`
if we want a default value from here
or `extras?.let {
view.updateText(it.getString(TEXT_EXTRA_KEY))
}`
if we only want to update when something passed in bundle

Reply:
I get your point but in this scenario I have to disagree. 
If there are no extras it because we used this fragment incorrectly.
We have method for instantiating it, so if extras are null then someone used that without that, so this is noop! 
Is should crash in the very first possible moment, to let developer/tester know about this.
Of course we could use ?: throw `SomethingMoreMeaningfull()` but in this case it's easy to debug without that.

Reply:
There is a reason why we're pushing for this. At very worst we're allowing for `?` operators at the domain levels. Reason being: using `!` and `?` leads to cutting corners. And it spreads, often forgoing the immutability and non-nullability. 

I get your point though. If passing the value is part of the contract , sure.  Though it often ends up with there is another code change, contract is broken, and defaulting logic is written anyways. Along the way, there is a user with a crashing app. Best case, tester with a crashing app.